functions/src/modules/SE.functions.ts

```diff
@@ -91,71 +91,68 @@ export const processNewAllotment = functions.database
 //
 /////////////////////////////////////////////////
 
-export const uploadProcessing = functions.storage.object()
+export const uploadProcessing = functions.storage.bucket(functions.config().sound_editing.restoration.uploads.bucket_name).object()
 .onFinalize(async object => {    
 
     const filePath = object.name;
-
-
-    if (!filePath.startsWith('sound-editing/restored/uploads'))
-        return -1;
+    const seUploadsBucket = admin.storage().bucket(functions.config().sound_editing.restoration.uploads.bucket_name);    
     
-    
-    const filePathRegex = /sound-editing\/restored\/uploads\/(\w+)\/(\w+)\/(\w+\/)*(([\w\d]+)-\d+-?\d+?)\.flac/;
+    const filePathRegex = /(\w+)\/(\w+\/)*(([\w\d]+)-\d+-\d+)\.flac/;
     
     const match = filePathRegex.exec(filePath);
     if (!match) {
-        console.log("Wrong Path -- File will be deleted.");
-        bucket.file(object.name).delete();
+        console.log(`Wrong Path -- ${filePath} will be deleted.`);
+        seUploadsBucket.file(object.name).delete();
```

Comment:
There are too many places where you delete the file. I mean in the code. Try to restructure the function in order to delete the file once in the end of processing. It may require to extract another function.

Reply:
Maybe throwing an error message from the validity checks would help also to log it in one `catch`.